lib/season_processor.rb

```diff
@@ -0,0 +1,82 @@
+module SeasonProcessor
+  def coach_stats(game_id_year, game_teams)
+    stats = Hash.new { |coach, stats| coach[stats] = [0.0, 0.0] }
+    game_teams.each do |game_team|
+      if game_team.game_id[0..3] == game_id_year
+        stats[game_team.head_coach][0] += 1
+        stats[game_team.head_coach][1] += 1 if game_team.result == "WIN"
+      end
+    end
+    return stats
+  end
+
+  def best_coach(coach_stats)
+    highest_win_percentage = 0.0
+    highest_win_percentage_coach = ""
+    coach_stats.each do |coach, stats|
+      if highest_win_percentage < (stats[1] / stats[0])
+        highest_win_percentage = stats[1] / stats[0]
+        highest_win_percentage_coach = coach
+      end
+    end
+    highest_win_percentage_coach
+  end
+
+  def worstest_coach(coach_stats)
+    lowest_win_percentage = Float::INFINITY
```

Comment:
I like the use of INFINITY for determining lowest_win_percentage, much more reliable than setting this equal to a "presumably" high number